joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/passes/PhpTypeRecovery.scala

```diff
@@ -0,0 +1,448 @@
+package io.joern.php2cpg.passes
+
+import io.joern.x2cpg.Defines
+import io.joern.x2cpg.passes.frontend._
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.{Operators, PropertyNames}
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.operatorextension.OpNodes
+import io.shiftleft.semanticcpg.language.operatorextension.OpNodes.{Assignment, FieldAccess}
+import overflowdb.BatchedUpdate.DiffGraphBuilder
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+class PhpTypeRecoveryPass(cpg: Cpg, config: XTypeRecoveryConfig = XTypeRecoveryConfig(iterations = 3))
+    extends XTypeRecoveryPass[NamespaceBlock](cpg, config) {
+
+  override protected def generateRecoveryPass(state: XTypeRecoveryState): XTypeRecovery[NamespaceBlock] =
+    new PhpTypeRecovery(cpg, state)
+}
+
+private class PhpTypeRecovery(cpg: Cpg, state: XTypeRecoveryState) extends XTypeRecovery[NamespaceBlock](cpg, state) {
+
+  override def compilationUnit: Iterator[NamespaceBlock] = cpg.file.namespaceBlock.iterator
+
+  override def generateRecoveryForCompilationUnitTask(
+    unit: NamespaceBlock,
+    builder: DiffGraphBuilder
+  ): RecoverForXCompilationUnit[NamespaceBlock] = {
+    val newConfig = state.config.copy(enabledDummyTypes = state.isFinalIteration && state.config.enabledDummyTypes)
+    new RecoverForPhpFile(cpg, unit, builder, state.copy(config = newConfig))
+  }
+}
+
+private class RecoverForPhpFile(cpg: Cpg, cu: NamespaceBlock, builder: DiffGraphBuilder, state: XTypeRecoveryState)
+    extends RecoverForXCompilationUnit[NamespaceBlock](cpg, cu, builder, state) {
+
+  override protected def prepopulateSymbolTable(): Unit = {
+    logger.debug(s"prepopulating symbol table")
+    super.prepopulateSymbolTable()
+    logger.debug(s"symbol table: ${symbolTable.view.mkString(" ")}")
+  }
+
+  override protected def prepopulateSymbolTableEntry(x: AstNode): Unit = x match {
+    case x: Call =>
+      x.methodFullName match {
+        case Operators.alloc =>
+        case _               => symbolTable.append(x, (x.methodFullName +: x.dynamicTypeHintFullName).toSet)
+      }
+    case _ => super.prepopulateSymbolTableEntry(x)
+  }
+
+  protected val methodTypesTable = mutable.Map[Method, mutable.HashSet[String]]()
+
+  override def isConstructor(c: Call): Boolean =
+    isConstructor(c.name) && c.code.endsWith(")")
+
+  override protected def isConstructor(name: String): Boolean =
+    !name.isBlank && name.charAt(0).isUpper
+
+  override def assignments: Iterator[Assignment] = {
+    logger.debug(
+      s"[assignments] assignments: ${cu.ast.isCall.nameExact(Operators.assignment).l.map(a => a.code).mkString("; ")}"
+    )
+    cu.ast.isCall.nameExact(Operators.assignment).map(new OpNodes.Assignment(_))
+  }
+
+  protected def unresolvedDynamicCalls: Iterator[Call] = cu.ast.isCall
+    .filter(_.dispatchType == "DYNAMIC_DISPATCH")
```

Comment:
If you import `io.shiftleft.codepropertygraph.generated.DispatchTypes` then use 
```suggestion
    .filter(_.dispatchType == DispatchTypes.DYNAMIC_DISPATCH)
```